src/components/password-modal/password-modal.tsx

```diff
@@ -1,163 +1,149 @@
-    ios: 20,
-    default: 0
-});
-
-export interface IProps {
-    visible: boolean;
-    infoText: string;
-    buttonLabel: string;
+export interface IExternalProps {
+    shouldCreatePassword?: boolean;
+    title?: string;
+    subtitle?: string;
     obRef: any;
 }
 
 interface IState {
-    password: string;
     visible: boolean;
-    revealPassword: boolean;
-    errorMessage: string;
+    title: string;
+    subtitle: string;
+    showTerms: boolean;
+    createPass: boolean;
+    verifyPass: boolean;
 }
 
 export class PasswordModalComponent extends React.Component<
-    IProps & IThemeProps<ReturnType<typeof stylesProvider>>,
+    IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>,
     IState
 > {
-    public onReject: any;
-    public onConfirm: any;
     private passwordRequestDeferred;
-
-    constructor(props: IProps & IThemeProps<ReturnType<typeof stylesProvider>>) {
+    constructor(props: IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>) {
         super(props);
 
         this.state = {
-            password: '',
-            revealPassword: false,
-            visible: !!props.visible || false,
-            errorMessage: ''
+            visible: false,
+            title: props.title ? props.title : translate('Password.pinTitleUnlock'),
+            subtitle: props.subtitle ? props.subtitle : translate('Password.pinSubtitleUnlock'),
+            showTerms: false,
+            createPass: false,
+            verifyPass: false
         };
-
         props.obRef && props.obRef(this);
     }
 
-    public requestPassword(): Promise<string> {
+    public async requestPassword(): Promise<string> {
         this.passwordRequestDeferred = new Deferred();
+
         this.setState({ visible: true });
+        const keychainPassword = await getPassword();
+        if (keychainPassword) {
+            this.setState({
+                showTerms: false
+            });
+        } else {
+            this.setState({
+                showTerms: true
+            });
+        }
+
         return this.passwordRequestDeferred.promise;
     }
 
     @bind
-    public onEnterPassword() {
-        // validate password
-        if (this.state.password !== 'pass') {
+    public async onPasswordEntered(value: string): Promise<string> {
+        if (this.state.createPass === true) {
             this.setState({
-                errorMessage: translate('Wallets.invalidPassword')
+                createPass: false,
+                verifyPass: true,
+                title: translate('Password.verifyPinTitle'),
+                subtitle: translate('Password.verifyPinSubtitle')
             });
-
             return;
         }
+        if (this.state.verifyPass === true) {
+            await setPassword(value);
+            this.passwordRequestDeferred && this.passwordRequestDeferred.resolve(value);
+            this.setState({
+                visible: false
+            });
+            return value;
```

Comment:
why do you return `value`, this would make `PasswordPin` component show an error message...